src/pages/home/report/ReportActionItemThread.tsx

```diff
@@ -1,62 +1,61 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Text, View} from 'react-native';
-import avatarPropTypes from '@components/avatarPropTypes';
 import MultipleAvatars from '@components/MultipleAvatars';
 import PressableWithSecondaryInteraction from '@components/PressableWithSecondaryInteraction';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
-import withWindowDimensions, {windowDimensionsPropTypes} from '@components/withWindowDimensions';
-import compose from '@libs/compose';
+import withWindowDimensions from '@components/withWindowDimensions';
+import {WindowDimensionsProps} from '@components/withWindowDimensions/types';
+import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@styles/useThemeStyles';
 import * as Report from '@userActions/Report';
 import CONST from '@src/CONST';
+import {Icon} from '@src/types/onyx/OnyxCommon';
 
-const propTypes = {
+type ReportActionItemThreadProps = WindowDimensionsProps & {
     /** List of participant icons for the thread */
-    icons: PropTypes.arrayOf(avatarPropTypes).isRequired,
+    icons: Icon[];
 
     /** Number of comments under the thread */
-    numberOfReplies: PropTypes.number.isRequired,
+    numberOfReplies: number;
 
     /** Time of the most recent reply */
-    mostRecentReply: PropTypes.string.isRequired,
+    mostRecentReply: string;
 
     /** ID of child thread report */
-    childReportID: PropTypes.string.isRequired,
+    childReportID: string;
 
     /** Whether the thread item / message is being hovered */
-    isHovered: PropTypes.bool.isRequired,
+    isHovered: boolean;
 
     /** The function that should be called when the thread is LongPressed or right-clicked */
-    onSecondaryInteraction: PropTypes.func.isRequired,
-
-    ...withLocalizePropTypes,
-    ...windowDimensionsPropTypes,
+    onSecondaryInteraction: () => void;
 };
 
-function ReportActionItemThread(props) {
+function ReportActionItemThread({numberOfReplies, icons, mostRecentReply, childReportID, isHovered, onSecondaryInteraction}: ReportActionItemThreadProps): React.ReactElement {
```

Comment:
AFAIK, we usually do not add a return type for a component. What's a reason here?